Quantum-Control-Applications/Superconducting/Single Flux Tunable Transmon/SSB mixer for readout/macros.py

```diff
@@ -10,6 +10,76 @@
 # QUA macros #
 ##############
 
+# Macro for performing a single shot active reset.
+def active_reset_single(threshold=None, Ig=None):
+    """Macro performing a single shot active reset.
+
+    :param threshold: threshold for the 'I' quadrature discriminating between ground and excited state.
+    :param Ig: A QUA variable for the information in the `I` quadrature. Should be of type `Fixed`. If not given, a new
+        variable will be created
+    :return: A QUA variable for the information in the `I` quadrature.
+    """
+    if Ig is None:
+        Ig = declare(fixed)
+    if threshold is None:
+        raise Exception("A threshold needs to be specified to perform active reset.")
+
+    measure(
+        "readout",
+        "resonator",
+        None,
+        dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", Ig),
+    )
+    # Perform active feedback
+    # Use the single conditional play statement for integrating active reset in other protocols
+    play("pi", "qubit", condition=(Ig < threshold))
+    return Ig
+
+
+# Macro for performing active reset until succesfull for a gicen number of tries.
+def active_reset_until_success(threshold=None, max_tries=1, Ig=None):
```

Comment:
I wouldn't give a default value to the threshold.
This makes it mandatory and then you don't need to check if it is None.

Reply:
Yes, done